rfcs/0014-wal_on_disk_format.md

```diff
@@ -0,0 +1,293 @@
+
+- [ ] Logical clocks
+- [ ] Time to live (TTL)
+- [ ] Compaction filters
+- [ ] Merge operator
+- [ ] Change Data Capture (CDC)
+
+<!-- TOC --><a name="metadata-coordination-and-lifecycles"></a>
+### Metadata, Coordination, and Lifecycles
+
+- [ ] Manifest format
+- [ ] Checkpoints
+- [ ] Clones
+- [ ] Garbage collection
+- [ ] Database splitting and merging
+- [ ] Multi-writer
+
+<!-- TOC --><a name="compaction"></a>
+### Compaction
+
+- [ ] Compaction state persistence
+- [ ] Compaction filters
+- [ ] Compaction strategies
+- [ ] Distributed compaction
+- [ ] Compactions format
+
+<!-- TOC --><a name="storage-engine-internals"></a>
+### Storage Engine Internals
+
+- [X] Write-ahead log (WAL)
+- [ ] Block cache
+- [ ] Object store cache
+- [ ] Indexing (bloom filters, metadata)
+- [ ] SST format or block format
+
+<!-- TOC --><a name="ecosystem-operations"></a>
+### Ecosystem & Operations
+
+- [ ] CLI tools
+- [ ] Language bindings (Go/Python/etc)
+- [ ] Observability (metrics/logging/tracing)
+
+<!-- TOC --><a name="operations"></a>
+## Operations
+
+<!-- TOC --><a name="performance-cost"></a>
+### Performance & Cost
+
+<!-- Describe performance and cost implications of this change. -->
+
+- **Latency (reads/writes/compactions):** 
+The latency for writes with durability guarantee mainly depends on the configured size of the encoded WAL object 
+(see `max_wal_bytes_size`) that is used to decide when the in-memory WAL data structure is flushed to object storage.
+Since the new persistence format can batch more records than the currently used SST format given the size of the 
+encoded WAL object, the latency might be higher.
+The latency for writes without durability guarantee might be lower if the in-memory data structure is a simple FIFO
+data structure as recommended in this RFC, since writes to the in-memory WAL are faster.
+A WAL object is read for recovery.
+If we define the latency of recovery as the time until the in-memory state is re-established then the new persistence 
+format might decrease the latency of recovery because more records can be read from object storage given an encoded size 
+of the WAL object.
+The laterncy of compaction is not affected by the new persistence format.
+- **Throughput (reads/writes/compactions):**
+The throughput of writes with durability guarantee might be higher since the new persistence format has less space 
+overhead without indices and filters. 
+Thus, with the new format slateDB can write more WAL records per time unit to object storage compared to the current 
+format given the same encoded size of the WAL object.
+Additionally, the new persistence format allows to recover more records per time unit.
+The throughput of compaction is not affected by the new persistence format.
+- **Object-store request (GET/LIST/PUT) and cost profile:**
+The new persistence format allows to store more records per PUT to and to get more records per GET from object storage. 
```

Comment:
If you're storing the wal on s3-express this change should reduce transfer costs (depending on the reduced overhead from indexes/filters)